### PR TITLE
LinuxRendererGLES: consolidate custom renderers

### PR DESCRIPTION
**Pull request by lrusak**
This is mostly dead code. Let's do what the GL renderer did.

The comment about CVREF render method indicates that this may break it.

### CODE REVIEW

**Comment by Memphiz** [comment]
So what is your Intention in Breaking CoreVideo Rendering? Not sure what I should review on that - that comment is there for a reason I am pretty sure. Now we have an extra y flip for corevideo references.

**Comment by Memphiz** [comment]
In have adapted my comment ...

**Comment by FernetMenta** [comment]
If vtb needs different code, it should be done in vtb, not generic gles. This was the idea of rework of renderers

**Comment by lrusak** [comment]
Of course my intention isn't to break it. I would like to fix it properly if it needs fixing.

**Comment by Memphiz** [comment]
@lrusak it doesn't compile:

http://jenkins.kodi.tv/job/IOS-ARM64/4782/console

in case you find an easy compile fix by looking at the log - please trigger jenkins for another test build for me to test. While trying to compile current master here yesterday - i realised that i need xcode9 now due to thread local storage changes. At that time my free time was over. ;)

File: xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.h [hunk 1]
```diff
@@ -64,13 +64,8 @@ struct YUVCOEF
 
 enum RenderMethod
 {
-  RENDER_GLSL   = 0x001,
-  RENDER_SW     = 0x004,
-  RENDER_POT    = 0x010,
```
Comment (Memphiz) [inline xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.h]: RENDER_POT is still referenced in CLinuxRendererGLES::CreateYV12Texture(int index) - thats why it doesn't compile ...

**Comment by Memphiz** [comment]
So i tried this by generating bookmarks while playing via VTB (using VideoCoreRef)

Nightly build: - bookmark is correctly oriented
This PR (+ the compile fix): bookmark looks correctly orientated. 
This PR (+ the compile fix) with the y flip commented out: upside down bookmark

So i guess something changed somewhere that makes the rotation mandatory for VTB / VCREF too. Just wondering why nightly had the correct orientation then - because it had the flip skipped for VCREF ...

All fine then i guess.

**Comment by lrusak** [comment]
@Memphiz really great! thanks for testing :+1:

**Review by FernetMenta — APPROVED** [review]

**Review by Memphiz — APPROVED** [review]
One once it compiles :)

**Pull request merged**